index.js

```diff
@@ -50,11 +53,12 @@ const factory = opts => {
         .then(cacheClient(iss))
         .then(thrush(kid))
 
-  const verify = curryN(2, partialRight(promisify(jwt.verify), [ verifyOpts ]))
+  const verify = pipe(stripBearer, curryN(2, partialRight(promisify(jwt.verify), [ verifyOpts ])))
 
   const authentic = token =>
     Promise.resolve(token)
       .then(tapP(enforce))
+      .then(stripBearer)
```

Comment:
The need to strip and null-check in multiple places is a 👃 that suggests we organize things a little better with some composition:

```js
// use unauthorized in enforce
const enforce = token =>
  token || Promise.reject(unauthorized('null token not allowed'))

const stripBearer = // <<-- move here to maintain alphebetic order
  replace(/^Bearer /i, '')

const unauthorized = err =>
  Promise.reject(Boom.wrap(err, 401))

const factory = opts => {
  // omiting other bits here...

  // remove stripBearer from verify
  const verify = curryN(2, partialRight(promisify(jwt.verify), [ verifyOpts ]))

  const authentic = token =>
    Promise.resolve(token)
      // remove tapP(enforce) here
      // remove stripBearer
      .then(decode)
      .then(tapP(checkIss))
      .then(getSigningKey)
      .then(chooseKey)
      .then(verify(token))
      .catch(unauthorized)

  // composeP to null-check and strip all in one place
  return composeP(authentic, stripBearer, tapP(enforce))
}
```